core/modules/replica/Controller.cc

```diff
@@ -232,681 +89,537 @@ string Controller::_context(string const& func) const {
+    return _submit<IndexRequest,
+                   decltype(database),
+                   decltype(chunk),
+                   decltype(hasTransactions),
+                   decltype(transactionId)>(
         workerName,
-        query,
-        user,
-        password,
-        maxRows,
-        [controller] (SqlRequest::Ptr request) {
-            controller->_finish(request->id());
-        },
+        database,
+        chunk,
+        hasTransactions,
+        transactionId,
+        onFinish,
         priority,
         keepTracking,
-        serviceProvider()->messenger()
-    );
-
-    // Register the request (along with its callback) by its unique
-    // identifier in the local registry. Once it's complete it'll
-    // be automatically removed from the Registry.
-
-    _registry[request->id()] =
-        make_shared<RequestWrapperImpl<SqlRequest>>(request, onFinish);
-
-    // Initiate the request
-
-    request->start(controller, jobId, requestExpirationIvalSec);
-
-    return request;
-}
-
-
-StopReplicationRequest::Ptr Controller::stopReplication(
-                                    string const& workerName,
-                                    string const& targetRequestId,
-                                    StopReplicationRequest::CallbackType const& onFinish,
-                                    bool keepTracking,
-                                    string const& jobId,
-                                    unsigned int requestExpirationIvalSec) {
-
-    LOGS(_log, LOG_LVL_DEBUG, _context(__func__) << "  targetRequestId = " << targetRequestId);
-
-    util::Lock lock(_mtx, _context(__func__));
-
-    return ControllerImpl::requestManagementOperation<StopReplicationRequest>(
-        shared_from_this(),
         jobId,
-        workerName,
-        targetRequestId,
-        onFinish,
-        keepTracking,
-        serviceProvider()->messenger(),
         requestExpirationIvalSec);
 }
 
 
-StopDeleteRequest::Ptr Controller::stopReplicaDelete(
-                                string const& workerName,
-                                string const& targetRequestId,
-                                StopDeleteRequest::CallbackType const& onFinish,
-                                bool keepTracking,
-                                string const& jobId,
-                                unsigned int requestExpirationIvalSec) {
-
-    LOGS(_log, LOG_LVL_DEBUG, _context(__func__) << "  targetRequestId = " << targetRequestId);
+SqlQueryRequest::Ptr Controller::sqlQuery(
+        string const& workerName,
+        string const& query,
+        string const& user,
+        string const& password,
+        uint64_t maxRows,
+        SqlQueryRequest::CallbackType const& onFinish,
+        int priority,
+        bool keepTracking,
+        string const& jobId,
+        unsigned int requestExpirationIvalSec) {
 
-    util::Lock lock(_mtx, _context(__func__));
+    LOGS(_log, LOG_LVL_DEBUG, _context(__func__));
```

Comment:
does this want to go in as DEBUG? Or can it be TRACE?

Reply:
and so forth for other log statements, I don't make any more noise about it.

Reply:
Thanks! I agree with you observation! I will migrate the log level.